server/src/main/java/io/pravega/schemaregistry/exceptions/CodecTypeNotRegisteredException.java

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.exceptions;
+
+/**
+ * Exception thrown when a codec type is not registered for the group. 
+ */
+public class CodecTypeNotRegisteredException extends RuntimeException {
```

Comment:
We should introduce a base exception that this exception extends from. That'd enable clients to deal with multiple types of exceptions in a similar way. Clients having to deal with all RuntimeException in the same way can cause exception  handling probelms.

Reply:
the client has a separate exceptions which are derived from a base class. Please look at `RegistryExceptions.java`.

These are server side exceptions that are only applicable on the service side business logic and are translated to appropriate HTTP error code.

Reply:
Even for server-side exceptions, it'd be good to have an exception hierarchy or at least a base exception from which all these SchemaRegistry specific exceptions extend. Just so that in places where these exceptions are handled within the server code itself, one can catch the base exception, rather than the `RuntimeException` which also represents programming errors. 

That being said, maybe you don't have a pressing need to do that right now as you haven't had to distinguish between the custom exceptions that extend from the RuntimeException directly and the rest of the RuntimeExceptions which may represent programming errors (like referencing a null).  In any case, this can also be done later as the need arises.